specs/protocol/granite/predeploys.md

```diff
@@ -8,9 +8,115 @@
+
+## Constants
+
+| Constant          | Value                           | Description |
+| ----------------- | ------------------------------- | ----------- |
+| `PREDEPLOY_COUNT` |  2048 | The total number of predeploys on L2 |
+| `L2ProxyAdmin` address | `0x4200000000000000000000000000000000000018` | The predeploy of the `ProxyAdmin` on L2 |
+| `DEPOSITOR_ACCOUNT` | `address(0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0001)` | Account that represents the system |
+| `COLLECTIVE_ACCOUNT` | `address(0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0002)` | Account that represents the collective |
+| `CHAIN_ACCOUNT` | `address(0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0003)` | Account that represents the chain operator |
+
+## System Predeploys
+
+System predeploys are predeploys that can only be modified via protocol upgrades. This means
+that the chain governance cannot modify them at the application layer. These predeploys
+are important for maintaining protocol level invariants and therefore the security of
+the system.
+
+TODO: define range of system predeploys
+
+## Userland Predeploys
+
+Userland predeploys are predeploys that can be modified by governance. They should not
+be responsible for managing the security of the system, meaning that if governance was
+to go rogue, then the system should still be considered secure.
+
+Userland predeploys are broken down into two categories, collective governed and chain
+governed.
+
+TODO: define range of userland predeploys, both collective and chain
+
+### Collective Governed
+
+The collective governed userland predeploys are managed at the superchain level. This means
+that the governance of the superchain as a whole is responsible for upgrading these predeploys.
+
+### Chain Governed
+
+The chain governed userland predeploys are managed on a per chain level. This is a commitment
+that these predeploys will never be used by the protocol or superchain governance and can be
+safely used by the chain operator for whatever purposes they desire.
+
+## L2ProxyAdmin
+
+The `L2ProxyAdmin` is a minimal proxy admin that is the ERC-1967 admin for each of the predeploys.
+For any permissioned call, the required caller is different depending on the address that is being
+called.
+
+To modify the implementation:
+- For system predeploys, it MUST be the `DEPOSITOR_ACCOUNT`
+- For collective governed predeploys, it MUST be the `COLLECTIVE_ACCOUNT`
+- For chain governed predeploys, it MUST be the `CHAIN_ACCOUNT`
+
+None of these accounts have known private keys, therefore they must come from specially crafted
+`TransactionDeposited` events that come from the `OptimismPortal`.
+
+## OptimismPortal
+
+A method for emitting special `TransactionDeposited` events is added to the `OptimismPortal`.
+Authorization must be set up such that the only the `SystemConfig` can result in a deposit
+transaction from the `CHAIN_ACCOUNT` and only the `SuperchainConfig` can result in a deposit
```

Comment:
This phrasing is unclear to me. Is the SuperchainConfig forwarding calls? If so, from whom?

Reply:
This is noted a bit in https://github.com/ethereum-optimism/specs/pull/194#discussion_r1605529214 but the idea is that the `SuperchainConfig` is now `Ownable` and in practice it would be the foundation owning it